graph/src/components/ethereum/types.rs

```diff
@@ -57,18 +57,44 @@ impl<'a, T> From<&'a Block<T>> for EthereumBlockData {
 #[derive(Clone, Debug)]
 pub struct EthereumTransactionData {
     pub hash: H256,
-    pub block_hash: H256,
-    pub block_number: U256,
+    pub index: U128,
+    pub from: H160,
+    pub to: H160,
+    pub value: U256,
     pub gas_used: U256,
+    pub gas_price: U256,
 }
 
 impl<'a> From<&'a Transaction> for EthereumTransactionData {
     fn from(tx: &'a Transaction) -> EthereumTransactionData {
         EthereumTransactionData {
             hash: tx.hash,
-            block_hash: tx.block_hash.unwrap(),
-            block_number: tx.block_number.unwrap(),
+            index: tx.transaction_index.unwrap(),
+            from: tx.from,
+            to: tx.to.unwrap_or(H160::from(0)),
```

Comment:
@leodasvacas Would it be realistic to support `ToAscObject<AscPtr<AscOption<U>>> for Option<T>` where `AscOption<U>` corresponds to `U | null` in AssemblyScript?

Reply:
Because I don't really like putting incorrect data like `0` in here.

Reply:
Agreed, I meant to present this as an open question in the description. :)

Reply:
This is straightforward to do. We'll want a `AscPtr::null()` constructor (just like the `default` impl). `EthereumTransactionData` should use `Option` for the optional fields. Then in `impl ToAscObj<AscEthereumTransaction> for EthereumTransactionData`, we should check for `None` and in that case use an `AscPtr::null()`, otherwise we call `heap.asc_new` on the `Some` payload.

Reply:
Great, let's do it. Thanks @leodasvacas!

Reply:
Done (pushed to this PR).